include/drivers/video.h

```diff
@@ -0,0 +1,556 @@
+	video_api_register_event_callback_t register_event_callback;
+};
+
+/**
+ * @brief Set video format.
+ *
+ * Configure video device with a specific format.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param ep Endpoint ID.
+ * @param fmt Pointer to a video format struct.
+ *
+ * @retval 0 Is successful.
+ * @retval -EINVAL If parameters are invalid.
+ * @retval -ENOTSUP If format is not supported.
+ * @retval -EIO General input / output error.
+ */
+static inline int video_set_format(struct device *dev,
+				   enum video_endpoint_id ep,
+				   struct video_format *fmt)
+{
+	const struct video_driver_api *api =
+		(const struct video_driver_api *)dev->driver_api;
+
+	__ASSERT(api->set_format, "set_format must be implemented by driver");
+
+	return api->set_format(dev, ep, fmt);
+}
+
+/**
+ * @brief Get video format.
+ *
+ * Get video device current video format.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param ep Endpoint ID.
+ * @param fmt Pointer to video format struct.
+ *
+ * @retval pointer to video format
+ */
+static inline int video_get_format(struct device *dev,
+				   enum video_endpoint_id ep,
+				   struct video_format *fmt)
+{
+	const struct video_driver_api *api =
+		(const struct video_driver_api *)dev->driver_api;
+
+	__ASSERT(api->get_format, "get_format must be implemented by driver");
+
+	return api->get_format(dev, ep, fmt);
+}
+
+/**
+ * @brief Enqueue a video buffer.
+ *
+ * Enqueue an empty (capturing) or filled (output) video buffer in the driver’s
+ * endpoint incoming queue.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param ep Endpoint ID.
+ * @param buf Pointer to the video buffer.
+ *
+ * @retval 0 Is successful.
+ * @retval -EINVAL If parameters are invalid.
+ * @retval -EIO General input / output error.
+ */
+static inline int video_enqueue(struct device *dev, enum video_endpoint_id ep,
+				struct video_buffer *buf)
+{
+	const struct video_driver_api *api =
+		(const struct video_driver_api *)dev->driver_api;
+
+	if (api->enqueue == NULL) {
+		return -ENOTSUP;
+	}
+
+	return api->enqueue(dev, ep, buf);
+}
+
+/**
+ * @brief Dequeue a video buffer.
+ *
+ * Dequeue a filled (capturing) or displayed (output) buffer from the driver’s
+ * enpoint outgoing queue.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param ep Endpoint ID.
+ * @param buf Pointer a video buffer pointer.
+ * @param timeout Timeout in milliseconds.
+ *
+ * @retval 0 Is successful.
+ * @retval -EINVAL If parameters are invalid.
+ * @retval -EIO General input / output error.
+ */
+static inline int video_dequeue(struct device *dev, enum video_endpoint_id ep,
+				struct video_buffer **buf,
+				u32_t timeout)
+{
+	const struct video_driver_api *api =
+		(const struct video_driver_api *)dev->driver_api;
+
+	if (api->dequeue == NULL) {
+		return -ENOTSUP;
+	}
+
+	return api->dequeue(dev, ep, buf, timeout);
```

Comment:
By hiding the IPC mechanism here your disallowing the use of k_poll. In the case where you have multiple cameras you'd like to recieve data from, with this API, you'd be polling each camera with this API using K_NO_WAIT, spinning. With k_poll you would get which video/camera device has a frame available as they come in. I would prefer to see something that is k_poll-able here.